### DISCUSSION TITLE
Merge vs squash

### DISCUSSION THREAD

**Discussion by thijssteel** [tlapack/tlapack#40] [General]
For larger PR's, I'm a big fan of squashing commits when merging. This means that on the main branch, we only see one commit from the entire PR. The has some major advantages:

- You don't need to focus as much on writing clean commit messages in your branch because they will not be a part of the permanent history
- While working on a larger piece of code, it is common to still have bugs that you solve later. This means that not every commit will actually work. If those commits make it onto the main branch, that is a bit ugly.
- The history is shorter and cleaner.

What are your opinions?

**Comment by weslleyspereira** [comment]
I like squashing, but I am not sure if I'd go for a single commit for every PR. I think the rate of single-commit PRs increase (or should increase) as the project becomes more mature. In new projects, like this one, I sometimes create a PR with one idea and then I end up improving related features. Sometimes a single commit of mine will have a list of changes, which I am completely aware it is not a good practice.

**Comment by weslleyspereira** [comment]
I recently noticed there is a Squash and Merge option on Github when merging a PR. I am trying to use it more frequently.
